src/main/java/xyz/nucleoid/plasmid/PlasmidConfig.java

```diff
@@ -0,0 +1,85 @@
+package xyz.nucleoid.plasmid;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParser;
+import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.JsonOps;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+public record PlasmidConfig(
+        Optional<String> userFacingPackAddress,
+        Optional<PlasmidWebServer.Config> webServerConfig
+) {
+    private static final Path PATH = Paths.get("config/plasmid.json");
+
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
+
+    private static final Codec<PlasmidConfig> CODEC = RecordCodecBuilder.create(instance ->
+        instance.group(
+                Codec.STRING.optionalFieldOf("resource_pack_address").forGetter(PlasmidConfig::userFacingPackAddress),
+                PlasmidWebServer.Config.CODEC.optionalFieldOf("webserver").forGetter(PlasmidConfig::webServerConfig)
```

Comment:
The field name should be consistent with the class naming:

```suggestion
                PlasmidWebServer.Config.CODEC.optionalFieldOf("web_server").forGetter(PlasmidConfig::webServerConfig)
```